### ISSUE TITLE
Inconsistent of training performance on kinetics400

### ISSUE THREAD

**Issue by Hanqer** [facebookresearch/TimeSformer#3]
Thanks for releasing the code. 
In my reproducing, I run your code  without modification except for my onw data (~24w training and ~1.975w validation). But get the performance inconsistent of your paper. 

I run the code with config 'TimeSformer_divST_8x32_224.yaml', the best validate top-1 accuracy after 15 epochs training only achieve   73.63%  which is much lower than the reported ~77%.  (8 x V100 GPUs, the running command is 
`python tools/run_net.py \
  --cfg configs/Kinetics/TimeSformer_divST_8x32_224.yaml \
  DATA.PATH_TO_DATA_DIR path_to_my_dataset \
  NUM_GPUS 8 \
  TRAIN.BATCH_SIZE 8 \` )

Are there some configuration different or something wrong with my setting?

**Comment by Hanqer** [comment]
You can find my training log via [Google Drive](https://drive.google.com/file/d/181_jHe_Dsbt0psry4RYfzfndTzjvuG1K/view?usp=sharing).

**Comment by gberta** [comment]
I looked at your training logs, and everything looks good. The 73.63% accuracy that you are getting is a clip-level accuracy obtained with random temporal sampling, and random spatial cropping. If you perform inference using "configs/Kinetics/TimeSformer_divST_8x32_224_TEST.yaml" configuration file, which does predictions by sampling the middle clip in a video, and doing 3 spatial crops, you should be able to reach the accuracy reported in the paper (i.e., ~77.7-78.0%).

Also, I believe that the full inference should have been performed at the end of training. Did that not happen?

**Comment by Hanqer** [comment]
@gberta  Thanks for your help. After testing, the top-1 accuracy seems consistent with the paper with 77.7%. 

But I wonder why the sampling rate and spatial crops have about 4% influence on top-1 accuracy?  As I am concerned, the length of the sampled video is 8 x 32 =256, which already cover a long range of the video. And the left middle right (top middle bottom) crops on spatial  vs. random crop on spatial.  Why these factors have huge impact on the accuracy? 

Besides,  I notice that a lazy optimizing is used in the training (64 iterations to accumulate the gradients with one optimization on the parameters). Does that equivalent to training with total batch size 64?

Thanks.

**Comment by gberta** [comment]
In my view, 4% boost is pretty reasonable in this case. The spatial center cropping is significantly more informative than random spatial cropping, and thus, it should yield a pretty substantial boost in performance. Also note that during training/validation, the videos are upsampled to a higher spatial resolution (i.e., [256-320], and therefore by using 224x224 crops you are effectively seeing less spatial context than you do during testing. Thus, I think 4% boost in the testing mode makes sense. 

As for gradient accumulation, I implemented it so that the model could be trained even on a single machine with GPUs that have less than 20GB of GPU memory. I did all of my experiments using 4 node training (with 32 32GB Tesla V100 gpus), which effectively made the batch size much larger. Researchers who don't have multiple machines, and who don't have 32GB GPUs won't be able to use similar batch size that I used. Thus, the gradient accumulation is meant to address this issue.
